### PR TITLE
11501 discoveredclusters rbac

### PR DESCRIPTION
**Pull request by zkayyali812**
https://github.com/open-cluster-management/backlog/issues/11501

### CODE REVIEW

File: frontend/src/routes/Credentials/Credentials.tsx [hunk 1]
```diff
@@ -107,6 +107,18 @@ export function CredentialsTable(props: {
...
+        if (props.discoveryConfigs) {
+            for (let i = 0; i < props.discoveryConfigs.length; i++) {
+                const discoNamespace = props.discoveryConfigs[i].metadata.namespace
+                if (discoNamespace === namespace) {
+                    return props.discoveryConfigs[i].metadata.name!
+                }
+            }
+        }
+        return ''
+    }
```
Comment (showeimer) [inline frontend/src/routes/Credentials/Credentials.tsx:120]:
```js
const discoveryNamespace = props.discoveryConfigs?.find((config) => config.metadata.namespace === namespace)
return discoveryNamespace ?? ''
```

**Review by cameronmwall — DISMISSED** [review]

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 2]
```diff
@@ -46,54 +58,59 @@ const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
...
+        cell: (discoveredCluster) =>
+            discoveredCluster?.spec.cloudProvider ? (
+                <AcmInlineProvider provider={getProvider(discoveredCluster?.spec.cloudProvider)} />
+            ) : (
+                '-'
+            ),
     },
     {
-        header: 'Last Active',
+        header: 'Last active',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:113]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 3]
```diff
@@ -46,54 +58,59 @@ const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
...
+                    <span key="openShiftVersion">{'OpenShift '.concat(discoveredCluster.spec.openshiftVersion)}</span>
+                )
+            } else {
+                return '-'
+            }
+        },
     },
     {
-        header: 'Infrastructure Provider',
+        header: 'Infrastructure provider',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:101]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 4]
```diff
@@ -46,54 +58,59 @@ const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
...
-        ),
+        header: 'Namespace',
+        sort: (a: DiscoveredCluster, b: DiscoveredCluster) =>
+            compareStrings(a?.metadata.namespace, b?.metadata.namespace),
+        search: (discoveredCluster) => discoveredCluster?.metadata.namespace ?? '-',
+        cell: (discoveredCluster) => discoveredCluster?.metadata.namespace ?? '-',
     },
     {
-        header: 'Distribution Version',
+        header: 'Distribution version',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:81]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 5]
```diff
@@ -46,54 +58,59 @@ const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
...
-        tooltip: 'TODO',
-        cell: (discoveredCluster) => (
-            <span key="connectedFrom">
-                &nbsp;{' '}
-                {discoveredCluster.spec.providerConnections === undefined
-                    ? ['N/A']
-                    : discoveredCluster.spec.providerConnections![0].name ?? 'N/A'}
-            </span>
-        ),
+        header: 'Namespace',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:74]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 6]
```diff
@@ -46,54 +58,59 @@ const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
         header: 'Status',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:58]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 7]
```diff
@@ -5,39 +5,51 @@ import {
...
 import { DiscoveredCluster } from '../../../resources/discovered-cluster'
 import { filterForProviderSecrets, ProviderConnection } from '../../../resources/provider-connection'
 import { useRecoilState } from 'recoil'
-import { DiscoveryConfig, DiscoveryConfigApiVersion, DiscoveryConfigKind } from '../../../resources/discovery-config'
+import { DiscoveryConfig } from '../../../resources/discovery-config'
 import { discoveredClusterState, discoveryConfigState, secretsState } from '../../../atoms'
 
 const discoveredClusterCols: IAcmTableColumn<DiscoveredCluster>[] = [
     {
         header: 'Name',
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:34]: translations needed

File: frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx [hunk 8]
```diff
@@ -261,105 +301,46 @@ export function DiscoveredClustersTable(props: {
...
-                                        confirm: () => {},
-                                        cancel: () => {},
-                                        title: '',
-                                        message: '',
-                                    })
-                                },
-                            })
-                        },
+        <AcmTable<DiscoveredCluster>
+            plural="discovered clusters"
```
Comment (jamestalton) [inline frontend/src/routes/Discovery/DiscoveredClusters/DiscoveredClusters.tsx:324]: translations needed

**Review by jamestalton — CHANGES_REQUESTED** [review]

**Comment by zkayyali812** [comment]
Closing due to soiling branch with bad DCO signoffs -  opened - https://github.com/open-cluster-management/console/pull/507 to contain fixes w/ proper signoffs

**Pull request closed without merging**